cloudflare.go

```diff
@@ -420,10 +420,20 @@ type reqOption struct {
 	params url.Values
 }
 
-// WithZoneFilter applies a filter based on zone name.
-func WithZoneFilter(zone string) ReqOption {
+// WithZoneFilters applies a filter based on zone properties.
+func WithZoneFilters(zoneName, accountID, status string) ReqOption {
```

Comment:
Looking through the documentation, `name`, `account.id` and `status` should provide enough flexibility here that we don't need to expand this again. If we do, it might be worth while replacing all of the `With*Filters` with the more generic method that accepts any `url.Values` instead and build the query from there instead.